grpc/shared_conn.go

```diff
@@ -75,8 +74,8 @@ type SharedConn struct {
 	// set to nil before this channel is closed.
 	peerConnFailed chan struct{}
 
-	resOnTrackMu  sync.Mutex
-	resOnTrackCBs map[resource.Name]OnTrackCB
+	onTrackCBByTrackNameMu sync.Mutex
```

Comment:
renameed variables as we are no longer storing resource names, but rather strings of the resource name